src/uniformscalingmap.jl

```diff
@@ -14,10 +14,17 @@ LinearAlgebra.issymmetric(::UniformScalingMap) = true
 LinearAlgebra.ishermitian(A::UniformScalingMap) = isreal(A)
 LinearAlgebra.isposdef(A::UniformScalingMap) = isposdef(A.λ)
 
+# comparison of UniformScalingMap objects, sufficient but not necessary
+Base.:(==)(A::UniformScalingMap, B::UniformScalingMap) = (eltype(A) == eltype(B) && A.λ == B.λ && A.M == B.M)
```

Comment:
I am starting to doubt whether we should compare `eltype` in `==`. Surely `Number` and `AbstractArray` instances can be equal without having the same `eltype`.